### PR TITLE
Add GuestNetwork Connected numbers sensor entity #790

### PR DESCRIPTION
**Pull request by GaryHuang-ASUS**
Add guest network connected devices number

### CODE REVIEW

File: custom_components/asusrouter/const.py [hunk 1]
```diff
@@ -1069,6 +1069,19 @@
...
+    # Connected GuestNetwork devices
+    ARSensorDescription(
+        key="gn_number",
+        key_group="devices",
+        name="Connected GuestNetwork Devices",
+        icon=ICON_ROUTER,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=True,
+        extra_state_attributes={
```
Comment (Vaskivskyi) [inline custom_components/asusrouter/const.py:1081]: We better remove the whole `devices` list from the state attributes for the guest network. It contains all the devices connected and will take too much database space

**Review by Vaskivskyi — CHANGES_REQUESTED** [review]
Hey, Gary. Thanks for your work! I have only one comment. Please, take a look. Otherwise, everything is good

File: custom_components/asusrouter/const.py (hunk 1 above)
Comment (GaryHuang-ASUS) [reply custom_components/asusrouter/const.py:1081]: Based on your suggestion, I have tried removing the `devices` list from `extra_state_attributes` and observed the changes in the `state_attributes` table in the database. Indeed, only the Connected Devices data remains updated(insert when device connected or disconnected).

Comment (GaryHuang-ASUS) [reply custom_components/asusrouter/const.py:1081]: By the way, I would like to know where these data are written into the database and how it is correctly retrieved and rendered to the frontend. 

While tracing the code, I only saw update_clients calling the API via the asusrouter(from ARDevice and call async_get_clients) package to fetch new data. The device's coordinator then updates its data attribute, but I didn't see where the data gets written to the database and what information(schema in the text part) is being recorded.

Thanks.

Comment (GaryHuang-ASUS) [reply custom_components/asusrouter/const.py]: By the way, I noticed that the loop was redundant, so I removed it and kept only the condition

Thanks.

Comment (Vaskivskyi) [reply custom_components/asusrouter/const.py]: All the DB-related tasks are handled by the Home Assistant itself. Integration only provides standardized entry points and interfaces.

E.g. when the integration is found by HA, it searches for the `async_setup_entry` function in [`__init__.py`](https://github.com/Vaskivskyi/ha-asusrouter/blob/dev/custom_components/asusrouter/__init__.py). As you can see, this one initializes a router (`ARDevice` instance) and tells HA what platforms to load (e.g. `sensor`, `switch` and so on). All those just use callbacks or defined HA function names for the communication.

But in general, integration does not have any access neither to the DB nor to the frontend

In principle, there are [HA dev docs](https://developers.home-assistant.io/docs/development_index). They help with most of the basics but do not have all the examples.

**Review by Vaskivskyi — APPROVED** [review]

**Comment by Vaskivskyi** [comment]
I think it's ready. Thanks! :+1:

**Pull request merged**